tools/cellxgene_census_builder/src/cellxgene_census_builder/build_soma/tissue_mapper.py

```diff
@@ -193,53 +51,3 @@ def reformat_ontology_term_id(ontology_term_id: str, to_writable: bool = True) -
             if ontology_term_id.count(":") != 1:
                 raise ValueError(f"{ontology_term_id} is an invalid ontology term id, it must contain exactly one ':'")
             return ontology_term_id.replace(":", "_")
-
-    def _list_ancestors(self, entity: owlready2.entity.ThingClass, ancestors: list[str] | None = None) -> list[str]:
-        """Recursive function that given an entity of an ontology, it traverses the ontology and returns
-        a list of all ancestors associated with the entity.
-        """
-        if ancestors is None:
-            ancestors = []
-
-        if self._is_restriction(entity):
-            # Entity is a restriction, check for part_of relationship
-
-            prop = entity.property.name
-            if prop != "BFO_0000050":
-                # BFO_0000050 is "part of"
-                return ancestors
-            ancestors.append(entity.value.name.replace("obo.", ""))
-
-            # Check for ancestors of restriction
-            self._list_ancestors(entity.value, ancestors)
-            return ancestors
-
-        elif self._is_entity(entity) and not self._is_and_object(entity):
-            # Entity is a superclass, check for is_a relationships
-
-            if entity.name in self.DENY_LIST:
```

Comment:
COG doesn't have a deny list filter. Is this a problem?

Reply:
Most likely. I was expecting this function to be fully replaceable by `get_high_level_terms`, if it's not then I think we're not ready for using it in the Census.

Reply:
we already filter out non-UBERON terms. I don't foresee issues with the two UBERON terms in the deny list, since they are high-level terms closer to the root node than anything in the curated tissue general list, so skipping fetching their ancestors will not affect traversing the graph used for mapping to one of the curated tissue general terms. I believe this would only become an issue if a term higher-level than "material anatomical entity" or "anatomical entity" is added to the curated tissue list--which doesn't fit the use-case of the list, and further wouldn't make sense if the goal is to filter out ancestor terms of these terms